viewing_party/party.py

```diff
@@ -1,23 +1,168 @@
-# -----------------------------------------
 
-# -----------------------------------------
-# ------------- WAVE 5 --------------------
-# -----------------------------------------
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
 
+    return user_data
+
+
+def watch_movie(user_data, movie_title):
+    watchlist = user_data["watchlist"]
+    watched = user_data["watched"]
+    for movie in watchlist:
+        if movie["title"] == movie_title:
+            watchlist.remove(movie)
+            watched.append(movie)
+
+    return user_data
+
+    # -----------------------------------------
+    # ------------- WAVE 2 --------------------
+    # -----------------------------------------
+
+
+def get_watched_avg_rating(user_data):
+    ratings_list = []
+    watched = user_data["watched"]
+
+    if len(watched) > 0:
+        for i in range(len(watched)):
+            ratings_list.append(watched[i]["rating"])
+    else:
+        return 0
+
+    ratings_sum = 0
+    for rating in ratings_list:
+        ratings_sum += rating
+
+    return ratings_sum/len(ratings_list)
+
+
+def get_most_watched_genre(user_data):
+    counts = {}
+    watched = user_data["watched"]
+
+    if len(watched) > 0:
+        for i in range(len(watched)):
+            genre = watched[i]["genre"]
+            counts[genre] = counts.get(genre, 0) + 1
+    else:
+        return None
+
+    return max(counts, key=counts.get)
+
+    # -----------------------------------------
+    # ------------- WAVE 3 --------------------
+    # -----------------------------------------
+
+# *******
+
+
+def get_unique_watched(user_data):
+    watched_list = user_data["watched"]
+    friends_watched_list = []
+    unique_list = []
+
+    for dict in user_data["friends"]:
+        for item in dict["watched"]:
+            friends_watched_list.append(item)
+
+    for movie in watched_list:
+        if movie not in friends_watched_list:
+            unique_list.append(movie)
+
+    return unique_list
+
+
+def get_friends_unique_watched(user_data):
+    watched_list = user_data["watched"]
+    friends_watched_list = []
+    unique_list = []
+
+    for dict in user_data["friends"]:
+        for item in dict["watched"]:
+            friends_watched_list.append(item)
+
+    for movie in friends_watched_list:
+        if movie not in watched_list and movie not in unique_list:
+            unique_list.append(movie)
+
+    return unique_list
+
+    # -----------------------------------------
+    # ------------- WAVE 4 --------------------
+    # -----------------------------------------
+
+
+def get_available_recs(user_data):
+    potential_recs_list = get_friends_unique_watched(user_data)
+    recs_list = []
+
+    for i in range(len(potential_recs_list)):
+        for sub in user_data["subscriptions"]:
+            if sub == potential_recs_list[i]["host"]:
+                recs_list.append(potential_recs_list[i])
+
+    return recs_list

```

Comment:
👍  Nice work using a helper function! How would this function look if we used a `for..in` loop?